SimeonInstaller.ps1

```diff
@@ -2797,6 +2806,309 @@ CRLFOption=CRLFAlways
+            [string]$Name = 'Simeon Sync',
+            [ValidateNotNullOrEmpty()]
+            [string]$GrantAccessToAppId
+        )
+        $token = Get-SimeonAzureADAccessToken -Resource 'PowerBI' -Tenant $Tenant
+        $restProps = @{
+            Headers = @{
+                Authorization = "Bearer $token"
+            }
+            ContentType = 'application/json'
+        }
+        $restPropsAdmin = @{
+            Headers = @{
+                Authorization = "Bearer $token"
+                'x-powerbi-user-admin' = $true
+            }
+            ContentType = 'application/json'
+        }
+
+        # Create the workspace if it doesn't exist
+        try {
+            $worksapce = (Invoke-WithRetry { Invoke-RestMethod @restProps "https://api.powerbi.com/v1.0/myorg/groups" }).value |? name -eq $Name
+        }
+        catch {
+            if ($_.ErrorDetails.Message -match 'User is not licensed') {
+                Write-Error "The current user is not licensed for Power BI, please assign a premium license and run the installer again"
+            } # TODO, what error does it throw if a user is not allowed to create workspaces, get that error type and throw that error
+            else {
+                Write-Error $_.Exception.Message
+            }
+        }
+
+        if (!$worksapce) {
+            $workspace = Invoke-WithRetry { Invoke-RestMethod @restProps "https://api.powerbi.com/v1.0/myorg/groups?workspaceV2=True" -Method Post -Body @"
+            {
+                "name":"$Name"
+            }
+"@ }
+        }
+        #TODO, does the report need premium capacity, if so need to add this:  "capacityObjectId":"E1C31DE5-E999-4990-9BCB-602F79CAADBD", (or capacityId)
+
+        # Update the tenant settings to allow Service Principals access to Power BI
+        # TODO Test with this URL, I made the change to west-us, but did not see that in the portal, even after waiting over night
+        $settings = Invoke-WithRetry { Invoke-RestMethod @restPropsAdmin 'https://wabi-us-west2-redirect.analysis.windows.net/metadata/tenantsettings' }
+        $servicePrincipalAccess = ($settings.featureSwitches |? switchName -eq 'ServicePrincipalAccess')
+        if (!$servicePrincipalAccess.isEnabled) {
+            $servicePrincipalAccess.isEnabled = $true
+
+            $body = @{}
+            $body['featureSwitches'] = @($servicePrincipalAccess)
+            $body['properties'] = @(
+                @{
+                    tenantSettingName = 'ServicePrincipalAccess'
+                    properties = @{
+                        HideServicePrincipalsNotification = "false"
+                    }
+                })
+            # TODO Power BI suggests only granting to certain groups. Should we create a group and add the app to it?
+            Invoke-WithRetry { Invoke-RestMethod @restPropsAdmin 'https://wabi-us-west2-redirect.analysis.windows.net/metadata/tenantsettings' -Method Put -Body (ConvertTo-Json -Depth 100 $body) }
```

Comment:
can it be assigned to multiple groups in case one already exists? what kind of group is it? AAD?